crates/router/src/connector/tsys.rs

```diff
@@ -227,7 +231,16 @@ impl ConnectorIntegration<api::PSync, types::PaymentsSyncData, types::PaymentsRe
         &self,
         req: &types::PaymentsSyncRouterData,
     ) -> CustomResult<Option<types::RequestBody>, errors::ConnectorError> {
-        let req_obj = tsys::TsysSyncRequest::try_from(req)?;
+
+        let connector_router_data = tsys::TsysRouterData::try_from((
+            &self.get_currency_unit(),
+            req.request.currency,
+            req.request.amount,
+            req,
+        ))?;
+
+        let req_obj = tsys::TsysPaymentsRequest::try_from(&connector_router_data)?;
+
```

Comment:
Undo these changes in `PaymentsSync` flow. 
This  implemented to be done for line `151` in the 

```
impl ConnectorIntegration<api::Authorize , types::PaymentsAuthorizeData, types::PaymentsResponseData>
    for Tsys
{
         ...
       fn get_request_body(
        &self,
        req: &types::PaymentsAuthorizeRouterData,
    ) -> CustomResult<Option<types::RequestBody>, errors::ConnectorError> {
                    ...
       }
}
```
and similar change has to be implemented for Capture flow, line `322`

```
impl ConnectorIntegration<api::Capture, types::PaymentsCaptureData, types::PaymentsResponseData>
    for Tsys
{
         ....
        fn get_request_body(
        &self,
        req: &types::PaymentsCaptureRouterData,
    ) -> CustomResult<Option<types::RequestBody>, errors::ConnectorError> {
        //do your changes here
          ...
     }
}